### ISSUE TITLE
Do not remove commented lines

### ISSUE THREAD

**Issue by rjking1** [cucumber/vscode#76]
### 👓 What did you see?

Formatting .feature file removes all commented lines 

### ✅ What did you expect to see?

commented lines are indented correctly but left intact

### 📦 Which tool/library version are you using?

latest at 16 May 2022

### 🔬 How could we reproduce it?

Comment a Given/etc line. It will be prefixed with a #
Choose to format the feature file
The commented out line will be removed !!

### 📚 Any additional context?

No. Makes it hard to keep this extension if it is going to remove my comments and commented out lines

**Comment by aslakhellesoy** [comment]
Thanks for reporting this @rjking1.

This will be fixed when https://github.com/cucumber/common/pull/1738 is fixed.

**Comment by JohnSmall** [comment]
double, triple and quadruple thumbs up on this one. I can't live with a coding tool that removes comments. It could at least ask.

**Comment by aslakhellesoy** [comment]
Hi @JohnSmall I agree it’s annoying. Do you want to help fix this?

**Comment by onthespotqa** [comment]
We ran into this issue and had to roll back having the wider org use the new extension because of this. Is there any status on a fix. We use the javascript implementation of cucumber.

**Comment by aslakhellesoy** [comment]
@onthespotqa it will be fixed when https://github.com/cucumber/common/pull/1738 is  fixed.

If this is important to your organisation, you can help by either submitting a new pull request or offer financialnsupport via the sponsor button.

Cucmber is maintained by volunteers. I have put a $1000 bounty on this issue.

**Comment by xeger** [comment]
Just a note that the new underlying issue to track is cucumber/gherkin-utils#16 -- it's equivalent to cucumber/common#1738 but in a new location.

**Comment by xeger** [comment]
There is an open PR under cucumber/gherkin-utils#16 which provides basic support for preserving comments. You can follow the conversation on that thread, but my understanding is that the volunteer Cucumber maintainers are overloaded right now -- they all have full-time day jobs -- and lack the time to review, QA or ship the fix.

The good news is, a fix (at least a partial fix) is nearly ready. I added some test cases for a common pattern within my org, which is to comment out an entire scenario/background/etc while debugging. It would be very beneficial if you could visit the PR and add any test cases in `testPretty.ts` that represent novel usage of comments in your org, so that we can ensure the VS Code extension will meet everyone's needs. If one of your test cases fails, I'll be happy to fix the product to make sure your comments get preserved.

**Comment by xeger** [comment]
**Short-term option:** my organization _really_ wants to adopt this extension. To help provide some QA for the eventual fix, and to unblock my organization, I have produced [a forked build of the Code extension](https://github.com/xeger/cucumber-vscode/releases/tag/v1.7.0) that you can install locally. (It is not published to the Code Marketplace, nor will it ever be.)

Other than preserving comments, this fork is identical to the official v1.7.0 extension that is published to the Visual Studio Code Marketplace.

You are all welcome to use this fork. If you have any issues, please don't report them to the Cucumber team -- they are not responsible for this fork. Instead, you can file issues directly with me, on my fork, or you can comment on [the corresponding gherkin-utils issue](https://github.com/cucumber/gherkin-utils/pull/16).
